pkg/internal/transform/spanner.go

```diff
@@ -149,6 +155,7 @@ func convertFromHTTPTrace(trace *ebpfcommon.HTTPRequestTrace) HTTPRequestSpan {
 		hostname = extractIP(trace.Host[:], int(trace.HostLen))
 	case EventTypeGRPCClient:
 		hostname, hostPort = extractHostPort(trace.Host[:])
+		peer = hostname
```

Comment:
Nice catch! I'm wondering if we are messing up the peer and host for HTTPClient, also I'm wondering is it legal to set hostname and peer to be the same for a gRPC client? Perhaps we can get RemoteAddr on the eBPF side?

Reply:
I'm not sure if we can get remoteAddr from the client-side calls. The way I understand it works (maybe I'm wrong) is that the connection is requested from the client side before the DNS are resolved, so from eBPF we can see the remote host but not the hostname.

Anyway I realized that this line is not needed with the current version of the code so I will remove it to avoid confusion, and we can decide later.